Add Article Container

jtart opened the pull request:
_Moves Article data logic to a container and introduces an Article presentational component._

NB: the `data` prop type is set to `any`. This is intentional until the MainContent [prop types are more defined](https://github.com/bbc/simorgh/issues/329)

* [x] Fixes https://github.com/bbc/simorgh/issues/323
* [x] Tests added for new features

* [ ] ~~Test engineer approval~~

Commit:
Move Article to containers

Commit:
Add Article container

Commit:
Move presentation to Article component

Commit:
Append Container to Article Container comp

Commit:
Add basic props

Commit:
Move Article story to component

jtart committed:
Merge branch 'latest' into article-container

Commit:
Name Article container as ArticleContainer

Commit:
Add temp any prop type for data

Commit:
Fill out Article stories

Commit:
Use textBlock helper so can add keys

Commit:
Add tests for Article comp

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into article-container

jtart committed:
Merge branch 'latest' into article-container

pjlee11 commented on src/app/components/Article/index.stories.jsx:
```diff
@@ -1,5 +1,78 @@
+      model: textBlock('This is a headline!'),
+    },
+    {
+      type: 'text',
```
We should open an issue to address the fact that we can't reuse the `textBlock()` function for this model similar to how the headline model is being defined. We would need to handle the model logic similar to how we've handled the propTypes duplication. 

For this particular method we could have something like: 
```
export const textBlock = (text, id, paragraphs = 1) => ({
const paragraphModel = paragraphs.map(text, index) => {
  {
    type: 'paragraph',
    blockId: `p-${index}`,
    model: {
      text,
    },
  },
};

blocks: [
    {
      type: 'text',
      blockId: `t-${id}`,
      model: {
        blocks: [
          paragraphModel
        ],
      },
    },
  ],
});
```

This way it's extendable to multiple paragraphs and multiple uses of the same method in a single model

jtart replied:
There is already a ticket to add fixture data from a library, see https://github.com/bbc/simorgh/issues/243. Let's start from there and see if that is appropriate.

pjlee11 commented on src/app/components/Article/index.stories.jsx:
```diff
@@ -1,5 +1,78 @@
+  ],
+};
+
+const filterData = filter => {
```
Could this be abstracted out into the helpers as it's likely it will be reused in the future

jtart replied:
I agree, but let's not abstract it out until there is more use-cases. Needless optimisation otherwise.

pjlee11 commented on src/app/containers/Article/index.jsx:
```diff
@@ -0,0 +1,48 @@
+  static async getInitialProps({ req, match } = {}) {
+    try {
+      const { id } = match.params;
+      let url = `/data/test/${id}.json`;
```
You'll need to remove `/test/` once latest is merged in

pjlee11 reviewed (changes requested):
Minor changes, nice work 😎

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into article-container

Commit:
Remove test from URL path

This work was already done in this PR https://github.com/bbc/simorgh/pull/316. However Article was moved to a container before the work was merged in and Git sees it as a new file, so the change wasn't picked up.

jtart committed:
Merge branch 'latest' into article-container

jtart committed:
Merge branch 'latest' into article-container

Commit:
Merge branch 'latest' of github.com:bbc/simorgh into article-container

Commit:
Remove hard-coded lang

Commit:
Rename back to lang and extract langauge from passport in Container
